source/slides/01_linux_basics.rst

```diff
@@ -0,0 +1,506 @@
+
+* Not just people: Apache, Mailman, ntp
+
+Users have
+----------
+
+* Username
+* UID
+* Group
+* Shell
+* Usually (but not always) password
+* Usually (but not always) home directory
+
+Managing users
+--------------
+
+.. code-block:: bash
+
+    $ cat /etc/passwd
+    # username:x:UID:GID:GECOS:homedir:shell
+    $ useradd $USER # vs adduser, the friendly Ubuntu version
+    $ userdel $USER
+    $ passwd
+
+.. figure:: ../_static/xkcd215.png
+    :align: center
+
+.. code-block:: bash
+
+    # GECOS: full name, office number and building, office phone extension,
+    # home phone number (General Electric Comprehensive Operating System)
+    $ chfn # change GECOS information; only works sometimes
+    $ finger # tells you someone's GECOS info
+
+Passwords
+---------
+
+* ``/etc/shadow``, not ``/etc/passwd``
+
+.. code-block:: bash
+
+    test@x230 ~ $ ls -l /etc/ | grep shadow
+    -rw-r-----  1 root shadow   1503 Nov 12 17:37 shadow
+
+    $ sudo su -
+    $ cat /etc/shadow
+    daemon:*:15630:0:99999:7:::
+    bin:*:15630:0:99999:7:::
+    sys:*:15630:0:99999:7:::
+    mail:*:15630:0:99999:7:::
+
+    # name:hash:time last changed: min days between changes: max days 
+    #    between changes:days to wait before expiry or disabling:day of
+    #    account expiry
+
+    $ chage # change when a user's password expires
+
+Root/Superuser
+--------------
+
+* UID 0
+* ``sudo``
+
+.. figure:: ../_static/xkcd149.png
+    :align: center
+
+Acting as another user
+----------------------
+
+.. code-block:: bash
+
+    $ su $USER          # become user, with THEIR password
+    $ su                # become root, with root's password
+    $ sudo su -         # use user password instead of root's
+    $ sudo su $USER     # become $USER with your password
+
+.. figure:: ../_static/xkcd838.png
+    :scale: 80%
+
+If someone has permissions errors:
+    * Check that they or their group owns the files
+    * Check that they have the flag +x to execute
+
+
+What are groups?
+----------------
+
+* Manage permissions for groups of users
+
+.. code-block:: bash
+
+    $ groupadd
+    $ usermod
+    $ groupmod
+    $ cat /etc/group
+        root:x:0:
+        daemon:x:1:
+        bin:x:2:
+        sys:x:3:
+        adm:x:4:
+        tty:x:5:
+    # group name:password or placeholder:GID:member,member,member
+
+What are files?
+---------------
+
+* Nearly everything
+* Files have:
+    * Owner
+    * Permissions
+    * inode
+    * Size
+    * Filename
+
+.. code-block:: bash
+
+    test@x230 ~ $ ls -il
+    total 8
+    2884381 drwxrwxr-x 5 test test 4096 Nov  6 11:46 Documents
+    2629156 -rw-rw-r-- 1 test test    0 Nov 13 14:09 file.txt
+    2884382 drwxrwxr-x 2 test test 4096 Nov  6 13:22 Pictures
+
+File extensions
+---------------
+
+* ``.jpg``, ``.txt``, ``.doc``
+
+* Really more of a recommendation
+    * File contains information about its encoding
+
+.. code-block:: bash
+
+    $ file $FILENAME # tells you about the filetype
```

Comment:
Might be worth mentioning it does this by magic file data, not by extension (both examples have extensions)